lib/ansible/plugins/cache/__init__.py

```diff
@@ -309,3 +309,61 @@ def _load(self, path):
 
     def _dump(self, value, path):
         return self._plugin._dump(value, path)
+
+
+class InventoryRedisCacheModule(BaseFileCacheModule):
```

Comment:
see #50446, we are going in the direction of generalizing the interface for all cache plugins, we don't want to have to create inventory specific cache plugins for every type